osgp/platform/osgp-adapter-ws-smartmetering/src/main/java/org/opensmartgridplatform/adapter/ws/smartmetering/application/services/MonitoringService.java

```diff
@@ -170,19 +98,37 @@ public String enqueueClearAlarmRegisterRequestData(@Identification final String
             @Identification final String deviceIdentification, final ClearAlarmRegisterRequest requestData,
             final int messagePriority, final Long scheduleTime) throws FunctionalException {
 
+        return enqueueRequestData(organisationIdentification, deviceIdentification, requestData, messagePriority,
+                scheduleTime, DeviceFunction.CLEAR_ALARM_REGISTER, MessageType.CLEAR_ALARM_REGISTER);
+    }
+    
+    public String enqueueActualPowerQualityRequestData(@Identification final String organisationIdentification,
+            @Identification final String deviceIdentification, final ActualPowerQualityRequest requestData,
+            final int messagePriority, final Long scheduleTime) throws FunctionalException {
+
+        return enqueueRequestData(organisationIdentification, deviceIdentification, requestData, messagePriority,
+                scheduleTime, DeviceFunction.GET_ACTUAL_POWER_QUALITY, MessageType.GET_ACTUAL_POWER_QUALITY);
+    }
+    
+    private String enqueueRequestData(@Identification final String organisationIdentification,
+            @Identification final String deviceIdentification, final Serializable requestData,
+            final int messagePriority, final Long scheduleTime, DeviceFunction deviceFunction, MessageType messageType) throws FunctionalException {
+
         final Organisation organisation = this.domainHelperService.findOrganisation(organisationIdentification);
         final Device device = this.domainHelperService.findActiveDevice(deviceIdentification);
 
-        this.domainHelperService.checkAllowed(organisation, device, DeviceFunction.CLEAR_ALARM_REGISTER);
+        this.domainHelperService.checkAllowed(organisation, device, deviceFunction);
 
-        LOGGER.debug("Enqueue clear alarm register request data called with organisation {} and device {}",
-                organisationIdentification, deviceIdentification);
+        if (log.isDebugEnabled()) {
```

Comment:
remove the if. that is really archaic.

Reply:
This is actually added because Sonar suggested it. Possibly to avoid unnecessary String construction